ssl/src/main/java/io/crate/protocols/postgres/ssl/SslReqHandlerSupplier.java

```diff
@@ -22,16 +22,31 @@
 
 package io.crate.protocols.postgres.ssl;
 
+import io.crate.settings.SharedSettings;
 import org.elasticsearch.common.settings.Settings;
 
+import java.lang.reflect.InvocationTargetException;
 import java.util.function.Supplier;
 
-public class SslHandlerProvider implements Supplier<SslReqHandler> {
+public class SslReqHandlerSupplier implements Supplier<SslReqHandler> {
 
     private final SslReqHandler sslReqHandler;
 
-    public SslHandlerProvider(Settings settings) {
-        this.sslReqHandler = new SslReqRejectingHandler(settings);
+    public SslReqHandlerSupplier(Settings settings) {
```

Comment:
Looks like this could be a single factory method to create the SSLReqHandler instead of creating a instance of the Supplier itself?